CMakeLists.txt

```diff
@@ -0,0 +1,113 @@
+        dependencies/c-utility/src/http_proxy_io.c
+        dependencies/c-utility/src/httpheaders.c
+        dependencies/c-utility/src/hmacsha256.c
+        dependencies/c-utility/src/hmac.c
+        dependencies/c-utility/src/xlogging.c
+        dependencies/c-utility/src/xio.c
+        dependencies/c-utility/src/ws_url.c
+        dependencies/c-utility/src/wsio.c
+        dependencies/c-utility/src/vector.c
+        dependencies/c-utility/src/uws_frame_encoder.c
+        dependencies/c-utility/src/uws_client.c
+        dependencies/c-utility/src/uuid.c
+        dependencies/c-utility/src/utf8_checker.c
+        dependencies/c-utility/src/sha384-512.c
+        dependencies/c-utility/src/sastoken.c
+        dependencies/c-utility/src/crt_abstractions.c
+        dependencies/c-utility/src/constbuffer_array_batcher.c
+        dependencies/c-utility/src/buffer.c
+        dependencies/umock-c/src/umocktypename.c
+        dependencies/umock-c/src/umockstring.c
+        dependencies/umock-c/src/umock_log.c
+        dependencies/umock-c/src/umock_c_negative_tests.c
+        dependencies/umock-c/src/umock_c.c
+        dependencies/umock-c/src/umockcallrecorder.c
+        dependencies/umock-c/src/umockcall.c
+        dependencies/umock-c/src/umocktypes_c.c
+        dependencies/umock-c/src/umocktypes.c
+        dependencies/umock-c/src/umocktypes_bool.c
+        dependencies/umock-c/src/umockcallpairs.c
+        dependencies/umock-c/src/umockautoignoreargs.c
+        dependencies/umock-c/src/umockalloc.c
+        dependencies/umock-c/src/umocktypes_wcharptr.c
+        dependencies/umock-c/src/umocktypes_stdint.c
+        dependencies/umock-c/src/umocktypes_charptr.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_ll.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_diagnostic.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_core_ll.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_authorization.c
+        azure-iot-sdk-c/iothub_client/src/iothub.c
+        azure-iot-sdk-c/iothub_client/src/iothub_module_client.c
+        azure-iot-sdk-c/iothub_client/src/iothub_device_client_ll.c
+        azure-iot-sdk-c/iothub_client/src/iothub_device_client.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_retry_control.c
+        azure-iot-sdk-c/iothub_client/src/iothub_transport_ll_private.c
+        azure-iot-sdk-c/iothub_client/src/iothubtransport.c
+        azure-iot-sdk-c/iothub_client/src/message_queue.c
+        azure-iot-sdk-c/iothub_client/src/iothubtransportmqtt.c
+        azure-iot-sdk-c/iothub_client/src/version.c
+        azure-iot-sdk-c/iothub_client/src/iothubtransport_mqtt_common.c
+        azure-iot-sdk-c/iothub_client/src/iothub_module_client_ll.c
+        azure-iot-sdk-c/iothub_client/src/iothub_message.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_edge.c
+        azure-iot-sdk-c/iothub_client/src/iothub_client_core.c
```

Comment:
Please also add the provisioning client (added in #11) to sources and includes, thanks

Reply:
@LDong-Arm I have added provisioning client source into CMake, please re-review